code/modules/antagonists/nightmare/nightmare.dm

```diff
@@ -1,5 +1,50 @@
 /datum/antagonist/nightmare
 	name = "Nightmare"
-	show_in_antagpanel = FALSE
-	show_name_in_check_antagonists = TRUE
+	show_in_antagpanel = TRUE
 	show_to_ghosts = TRUE
+	job_rank = ROLE_NIGHTMARE
+	antagpanel_category = "Nightmare"
+	show_name_in_check_antagonists = TRUE
+
+/datum/antagonist/nightmare/on_gain()
+	forge_objectives()
+	return  ..()
+
+/datum/antagonist/nightmare/greet()
+	owner.announce_objectives()
+	to_chat(owner, "<span class='boldannounce'>Your primary goal is keeping the station dark, do not go out of your way to randomly kill people. \
```

Comment:
This should probably be a little bit more IC, as in "You are a defender of the darkness, you don't want to kill, only snuff out lights and defend yourself" or something like that. if this is copied from somewhere it can be left alone though

Reply:
> if this is copied from somewhere it can be left alone though
Suggested by one of the admins - Ro